tools/setup_perf_cluster.sh

```diff
@@ -228,6 +228,95 @@ function run_fortio_test3() {
   Execute curl "http://$VM_IP/fortio/?json=on&qps=-1&t=30s&c=48&load=Start&url=http://$ISTIO_INGRESS_IP/fortio1/echo"
 }
 
+# Run canonical perf tests.
+# The following parameters can be supplied:
+# 1) Label:
+#    A custom label to use. This is useful when running the same suite against two target binaries/configs.
+#    Defaults to "canonical"
+# 2) Driver:
+#    The load driver to use. Currently "fortio1" and "fortio2" are supported. Defaults to "fortio1".
+# 3) Target:
+#     The target service for the load. Currently "echo1" and "echo2" are supported.
+#     Defaults to "echo2"
+# 4) QPS:
+#     The QPS to apply. Defaults to 400.
+# 5) Duration:
+#     The duration of the test. Default is 5 minutes.
+# 6) Clients:
+#     The number of clients to use. Defaults is 8.
+# 7) Outdir:
+#     The output dir for collecting the Json results. If not specified, a temporary dir will be created.
+function run_canonical_perf_test() {
+    LABEL="${1}"
+    DRIVER="${2}"
+    TARGET="${3}"
+    QPS="${4}"
+    DURATION="${5}"
+    CLIENTS="${6}"
+    OUT_DIR="${7}"
+
+    # Set defaults
+    LABEL="${LABEL:-canonical}"
+    DRIVER="${DRIVER:-fortio1}"
+    TARGET="${TARGET:-echo2}"
+    QPS="${QPS:-400}"
+    DURATION="${DURATION:-5m}"
+    CLIENTS="${CLIENTS:-8}"
+
+    get_istio_ingress_ip
+
+    FORTIO1_URL="http://${ISTIO_INGRESS_IP}/fortio1/fortio"
+    FORTIO2_URL="http://${ISTIO_INGRESS_IP}/fortio2/fortio"
+    case "${DRIVER}" in
+        "fortio1")
+            DRIVER_URL="${FORTIO1_URL}"
+            ;;
+        "fortio2")
+            DRIVER_URL="${FORTIO2_URL}"
+            ;;
+        *)
+            echo "unknown driver: ${DRIVER}"
+            exit -1
+            ;;
+    esac
+
+    # URL encoded URLs for echo1 and echo2. These get directly embedded as parameters into the main URL to invoke
+    # the test.
+    ECHO1_URL="http%3A%2F%2Fechosrv1%3A8080%2Fecho%3Fdelay%3D250us%3A30%25%2C5ms%3A5%25%26status%3D503%3A0.5%25%2C429%3A1.5%25"
+    ECHO2_URL="http%3A%2F%2Fechosrv2%3A8080%2Fecho%3Fdelay%3D250us%3A30%25%2C5ms%3A5%25%26status%3D503%3A0.5%25%2C429%3A1.5%25"
+    case "${TARGET}" in
+        "echo1")
+            TARGET_URL="${ECHO1_URL}"
+            ;;
+        "echo2")
+            TARGET_URL="${ECHO2_URL}"
+            ;;
+        *)
+            echo "unknown target: ${TARGET}"
+            exit -1
+            ;;
+    esac
+
+    PERCENTILES="50%2C+75%2C+90%2C+99%2C+99.9"
```

Comment:
you shouldn't need to specify this (like you don't need to when using the command line directly) but right now you do, filled https://github.com/istio/fortio/issues/164 for that so for now you can keep it (ditto about url escaping though, it works without it)

Reply:
Unfortunately, this is still needed. % is an escape character, and without percent sign, I can't make things work.

I think UI needs a bit more polish, but otherwise things seems to be pretty useful.

One thing worth adding probably would be around errors. There probably should be an error rate threshold.